src/server.c

```diff
@@ -4497,19 +5076,26 @@ int processCommand(client *c) {
 
     /* Now lookup the command and check ASAP about trivial error conditions
      * such as wrong arity, bad command name and so forth. */
-    c->cmd = c->lastcmd = lookupCommand(c->argv[0]->ptr);
+    c->cmd = c->lastcmd = lookupCommand(c->argv,c->argc);
     if (!c->cmd) {
-        sds args = sdsempty();
-        int i;
-        for (i=1; i < c->argc && sdslen(args) < 128; i++)
-            args = sdscatprintf(args, "`%.*s`, ", 128-(int)sdslen(args), (char*)c->argv[i]->ptr);
-        rejectCommandFormat(c,"unknown command `%s`, with args beginning with: %s",
-            (char*)c->argv[0]->ptr, args);
-        sdsfree(args);
+        if (lookupCommandBySds(c->argv[0]->ptr)) {
+            /* If we can't find the command but argv[0] by itself is a command
+             * it means we're dealing with an invalid subcommand. Print Help. */
+            addReplySubcommandSyntaxError(c);
+        } else {
+            sds args = sdsempty();
+            int i;
+            for (i=1; i < c->argc && sdslen(args) < 128; i++)
+                args = sdscatprintf(args, "'%.*s' ", 128-(int)sdslen(args), (char*)c->argv[i]->ptr);
+            rejectCommandFormat(c,"unknown command '%s', with args beginning with: %s",
+                (char*)c->argv[0]->ptr, args);
+            sdsfree(args);
+        }
         return C_OK;
     } else if ((c->cmd->arity > 0 && c->cmd->arity != c->argc) ||
-               (c->argc < -c->cmd->arity)) {
-        rejectCommandFormat(c,"wrong number of arguments for '%s' command",
+               (c->argc < -c->cmd->arity))
+    {
```

Comment:
it was a multiline `if`, the old style was better (and also no need to change these lines (blame log))

Reply:
we usually drop a line if the `if` is more than one line

either we leave it like this or I will have the `if` condition in one line (anyway it will change the blame log)

unless you insist

Reply:
but there is more than one line in the `if` condition. and i don't see you changed anything in that line, so why would it change?

Reply:
because i saw this violation of the Redis coding standard, so I fixed it...

Reply:
Ohh, sorry, I think I looked at it in the wrong direction, I need the new one is the right one.